src/Adapter/Store/AbstractStoreHandler.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Store;
+
+use PrestaShop\PrestaShop\Core\Domain\Store\Exception\StoreNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Store\ValueObject\StoreId;
+use Store;
+use Validate;
+
+abstract class AbstractStoreHandler
+{
+    protected function getStore(StoreId $storeId): Store
```

Comment:
I have one suggestion about this one, although it works fine and is totally acceptable We recently introduced (in the product page handlers) the notion of Repositories that we try to use as our default architecture in the handlers from now on

Would you be motivated to use it as well in this PR? We created an abstract class to help us build them, these are repositories that are based on the use of legacy ObjectModel classes. The abstract class provides a few protected methods that can be used to build the public method of each repository.

Here is an example https://github.com/PrestaShop/PrestaShop/blob/develop/src/Adapter/Currency/Repository/CurrencyRepository.php

The idea is to centralize the access to object model via these repositories to avoid accessing object models direct This will help us later when we need to migrate to another persistence ORM (maybe Doctrine maybe something else)

Reply:
In your case, you can already use the `StoreNotFoundException` which is already present (even if it's not used which is weird ^^)

Reply:
So if you use the repository architecture, you would inject the `StoreRepository` inside your handlers instead of relying on an abstract class This way the code is more reusable as well

It would need to be created into the `PrestaShop\PrestaShop\Core\Domain\Store\Repository` namespace (we plan on moving the `Adapter` services into the `Core` namespace but here you create it in the appropriate namespace from the beginning)

Reply:
Ok i see, will try to make it asap